### ISSUE TITLE
10_1_X IB failure in wf 20261.98

### ISSUE THREAD

**Issue by fabiocos** [cms-sw/cmssw#22660]
After the last updated of test workflows, there is a DAS error in the wf 20261.98 due to an apparently missing input file in the search

dasgoclient --limit 0 --query 'file dataset=/RelValNuGun/CMSSW_9_3_2-93X_upgrade2023_realistic_v2_2023D17noPU-v1/GEN-SIM site=T2_CH_CERN'

**Comment by fabiocos** [comment]
assign pdmv

**Comment by fabozzi** [comment]
@kpedro88 apparently this GEN-SIM does not exists. How to proceed with that? Produce it now or wait for the next round of Phase 2 relvals?

**Comment by kpedro88** [comment]
@fabozzi @makortel for 10_1_X I think we should just produce it with 10_1_0_pre3 relvals

**Comment by makortel** [comment]
Hmm, it worked in the PR test of #22510. Could the difference be that now that `defaultDataSets['2023D17']` is set, also the step1 GEN-SIM gets replaced by query to DAS in IB tests? (are they are run with ` runTheMatrix.py --i all`?)

Alternatively the workflow could be instructed to always run NuGun GEN-SIM (AFAICT that's what happens with run2 workflows `250199.*`, so maybe that's what should be done for phase2 as well)

**Comment by makortel** [comment]
To clarify, what the workflow really needs via DAS is the MinBias GEN-SIM for pileup input, and that one does exist for 9_3_2.

On the other hand, I don't object producing phase2 RelVals in 10_1_0_pre3 (on the contrary).

**Comment by makortel** [comment]
IIUC, this

> Alternatively the workflow could be instructed to always run NuGun GEN-SIM (AFAICT that's what happens with run2 workflows 250199.*, so maybe that's what should be done for phase2 as well)

would mean to instead of setting `defaultDataSets['2023D17']`
https://github.com/cms-sw/cmssw/blob/fe62f5a601dfe5f13bd4154ac9c15f0bf10511d3/Configuration/PyReleaseValidation/python/relval_steps.py#L2248
set `PUDataSets['2023D17']` explicitly.
https://github.com/cms-sw/cmssw/blob/fe62f5a601dfe5f13bd4154ac9c15f0bf10511d3/Configuration/PyReleaseValidation/python/relval_steps.py#L2280

@kpedro88 @fabiocos Let me know which you prefer, RelVals in 10_1_0_pre3 or drop the requirement on `NuGun` GEN-SIM dataset input.

**Comment by kpedro88** [comment]
@makortel would setting `PUDataSets['2023D17']` change the input for classical mixing as well?

**Comment by makortel** [comment]
> would setting `PUDataSets['2023D17']` change the input for classical mixing as well?

It would. If that is a problem, we could make additional `PUPRMXDataSets` to hold only that.

I see that disabling the automated input for premixing stage1 would be fairly straightforward as well. 

I think there are actually two issues
1. with `-i all` 20261.98 requires NuGun GEN-SIM, whereas run2 premixing stage1 workflows do not
   * I suppose it would be good to make 20261.98 similar to run2 stage1 workflows in this respect. I believe this would be sufficient to get the workflow running.
2. how to configure the MinBias dataset for pileup input
   * `defaultDataSets`? 
   * `PUDataSets`?
   * premixing-specific `PUDataSets`?

**Comment by kpedro88** [comment]
Yes, if point 1 is a difference between run2 and phase2, let's make them agree.

Maybe there should be a premixing-specific PUDataSet for Phase2. I think it's the simplest way to keep things from interfering with each other.

**Comment by makortel** [comment]
I made a PR #22677 to address the point 1 and thus making the 20261.98 workflow to run with `-i all`.

I did not do anything yet for point 2 as I'm not sure yet on the best strategy.

For run2 premixing stage1 the pileup dataset comes from `baseDataSetRelease`
https://github.com/cms-sw/cmssw/blob/fe62f5a601dfe5f13bd4154ac9c15f0bf10511d3/Configuration/PyReleaseValidation/python/relval_steps.py#L1259
https://github.com/cms-sw/cmssw/blob/fe62f5a601dfe5f13bd4154ac9c15f0bf10511d3/Configuration/PyReleaseValidation/python/relval_steps.py#L1123
https://github.com/cms-sw/cmssw/blob/fe62f5a601dfe5f13bd4154ac9c15f0bf10511d3/Configuration/PyReleaseValidation/python/relval_steps.py#L520
which is then the same GEN-SIM as is used for classical mixing pileup (and the same pattern as for signal GEN-SIM).

Corresponding treatment for the upgrade matrix would be to continue to set `defaultDataSets`, and let premixing stage1 to use the same pileup input as classical mixing. (stage2 pileup input would need a separate treatment when we get there)

Then we can of course discuss if there are other reasons to separate the pileup input of premixing from the others along

> I think it's the simplest way to keep things from interfering with each other.

so how would setting `defaultDataSets['2023D17']` to 9_3_2 GEN-SIM interfere with anything? I can think of the following side effects
* with `-i all` signal GEN-SIM is taken from 9_3_2 instead of regeneration
* classical mixing workflows would actually work since they would have pileup input

Which of these would be undesired? Or is there something I did not consider?

**Comment by kpedro88** [comment]
I was actually thinking about stage2, not stage1. It should be fine to use the same input for stage1 and classical mixing.

**Comment by makortel** [comment]
Ok. So let's continue with stage1 as it is, and add the additional dictionary for stage2 at some point.
